From 76720b56a8c953bb9f3228dbd8219ede0a4b096e Mon Sep 17 00:00:00 2001
From: Yann Dirson <yann.dirson@vates.tech>
Date: Mon, 8 Dec 2025 16:37:59 +0100
Subject: [PATCH 2/4] updater.py: switch to python3

Despite the unversioned "python" shebang, the code was indeed py2-specific.

Signed-off-by: Yann Dirson <yann.dirson@vates.tech>
---
 SOURCES/etc/xapi.d/plugins/updater.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/SOURCES/etc/xapi.d/plugins/updater.py b/SOURCES/etc/xapi.d/plugins/updater.py
index 1140207..b296bb7 100755
--- a/SOURCES/etc/xapi.d/plugins/updater.py
+++ b/SOURCES/etc/xapi.d/plugins/updater.py
@@ -1,8 +1,8 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 
 from functools import wraps
-import ConfigParser
+import configparser
 import json
 import yum
 
@@ -239,7 +239,7 @@ CONFIGURATION_FILE = '/etc/yum.repos.d/xcp-ng.repo'
 @error_wrapped
 @operationlock()
 def get_proxies(session, args):
-    config = ConfigParser.ConfigParser({'proxy': '_none_'})
+    config = configparser.ConfigParser({'proxy': '_none_'})
     config.read(CONFIGURATION_FILE)
     new_dict = dict((section, config.get(section, 'proxy', '_none_')) for section in config.sections())
     return json.dumps(new_dict)
@@ -257,7 +257,7 @@ def set_proxies(session, args):
     special_url_prefix = 'http://HTTPS///'
     https_url_prefix = 'https://'
     proxies = json.loads(args['proxies'])
-    config = ConfigParser.ConfigParser()
+    config = configparser.ConfigParser()
     if CONFIGURATION_FILE not in config.read(CONFIGURATION_FILE):
         raise Exception('could not read file %s' % CONFIGURATION_FILE)
 
-- 
2.47.3

